bus-mapping/src/evm/opcodes/codecopy.rs

```diff
@@ -69,7 +69,7 @@ fn gen_copy_event(
         state.gen_copy_steps_for_bytecode(exec_step, &bytecode, src_addr, dst_addr, length)?;
 
     Ok(CopyEvent {
-        src_type: CopyDataType::Bytecode,
+        src_type: CopyDataType::Bytecode(Default::default()),
```

Comment:
tried similar solution at earlier time when starting to implement byte code refractor:  added is_first_bytecode info  into CopyEvent and set it to correct value in block_convert method. but what not good of the solution are: 1.  enlarge changing scope, copy event related files (like files under `evm/opcodes` folder)
2.  more complex code: first time CopyDataType::Bytecode not correct,  correct it in different place. 
so later changed to remain CopyEvent the same, only get the bytecode information when required(CopyTable::assignment).

Reply:
> first time CopyDataType::Bytecode not correct, correct it in different place.
I think "fixing" it in block_convert is fine, because that's also the place where we partition the bytecodes. we partition the bytecodes at https://github.com/scroll-tech/zkevm-circuits/pull/1402/files#diff-e71dae4a71b5661aaa2385201c4554b7d56790191c6e5d0a54ebdcac4c63dd92R593 and then two lines after, we update the copy events.

Reply:
if we can avoid to set it twice, would that be better?

Reply:
yes, that would be better. i'm not sure it's possible though, since the bus code mapping can't know which tables the bytecodes will end up in until the whole block has been seen.

Reply:
keep CopyEvent unchanged and only get the bytecode table information when required(CopyTable::assignment).